### PR TITLE
CI: Use sccache action from marketplace

### PR DESCRIPTION
**Pull request by mwestphal**
Thanks for contributing to TTK!

Before submitting your pull request, please:

- [x] Review our [Contributor Guidelines](https://github.com/topology-tool-kit/ttk/blob/dev/CONTRIBUTING.md), in particular regarding code formatting (with clang-format) and continuous integration.

- [x] Please provide a quick description of your contributions below:

Remove ccache usage and use sscache marketplace action instead.
https://github.com/marketplace/actions/install-and-setup-sccache

### CODE REVIEW

**Comment by mwestphal** [comment]
Caches have been created as expected: https://github.com/topology-tool-kit/ttk/actions/caches

**Comment by mwestphal** [comment]
@julien-tierny @CharlesGueunet ptal

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -67,6 +64,13 @@ jobs:
           libqhull-dev \
           dpkg-dev
         sudo python3 -m pip install scikit-learn
+        cargo install sccache --version 0.4.2 --locked
```
Comment (mwestphal) [inline .github/workflows/test.yml:67]: ideally this (and all the apt install above) should be done in a Dockerfile instead of doing this in CI

**Comment by julien-tierny** [comment]
thanks a lot @mwestphal for the PR!
it looks all good to me.
@pierre-guillou what do you think?

**Comment by pierre-guillou** [comment]
It looks very nice and definitely simpler that what I've built.

File: .github/workflows/test.yml (hunk 1 above)
Comment (pierre-guillou) [reply .github/workflows/test.yml:67]: Wouldn't it be quicker to fetch binary archives directly from https://github.com/mozilla/sccache/releases/tag/v0.4.2?

Comment (mwestphal) [reply .github/workflows/test.yml:67]: cargo install take some time but thats still faster than windows CI anyway. A single line install is better than installing it manually imo.

**Comment by mwestphal** [comment]
Let me know If I need to rebase.

**Comment by julien-tierny** [comment]
@mwestphal 
that looks all good. I've just resolved the conflict.
can you please double check I haven't broken anything? thanks!

**Comment by mwestphal** [comment]
> @mwestphal that looks all good. I've just resolved the conflict. can you please double check I haven't broken anything? thanks!

Looks good to me

**Comment by julien-tierny** [comment]
ok, let's go!
thanks a lot!

**Pull request merged**